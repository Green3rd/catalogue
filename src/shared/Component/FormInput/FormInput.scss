@import "../../common.scss";

.FormInput {
  display: flex;
  flex-direction: column;
  padding: 8px 0;
  align-items: flex-start;
  width: 100%;

  @mixin inputStyle($important) {
    font-size: inherit $important;
    padding: 6px 12px $important;
    margin: 6px 0;
    width: 94% $important;
    border-radius: 4px $important;
    border: 1px solid #646464 $important;
    transition: border 0.15s ease-in-out 0s, box-shadow 0.15s ease-in-out 0s;
    height: auto $important;

    &:focus {
      border-color: rgb(201, 201, 201) $important;
      outline: none;
      box-shadow: $boxShadow;
    }

    &--error {
      border-color: $mainColor;
    }

    @include for-tablet-landscape-up {
      width: 96% $important;
    }

    @include for-phone-only {
      width: 91% $important;
    }
  }

  &__Label {
  }

  &__Field {
    @include inputStyle(null);

    &--noArrow {
      -moz-appearance: textfield;
      &::-webkit-inner-spin-button {
        display: none;
      }
      &::-webkit-outer-spin-button,
      &::-webkit-inner-spin-button {
        -webkit-appearance: none;
        margin: 0;
      }
    }
  }

  &__Error {
    color: $mainColor;
    font-size: 0.7em;
  }

  // For InputAddress
  &--children {
    // Container
    label + div {
      width: 100%;
      div:only-child {
        width: 100%;
      }
    }

    // suggestion box container
    input + div {
      min-width: 300px !important;

      // Item
      div {
        border-radius: 4px;
        padding: 4px 16px !important;
        color: $hoverBackground;
        font-size: 0.8em;
      }
    }

    input {
      @include inputStyle(!important);
    }

    &.FormInput--error {
      input {
        border-color: $mainColor !important;
      }
    }
  }
}
