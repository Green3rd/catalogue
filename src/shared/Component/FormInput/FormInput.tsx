import React from "react";
import "./FormInput.scss";
import { Field, ErrorMessage } from "formik";
import { generateClassName } from "shared/utils/reactClassName";
import { ReactElement } from "react";

interface Props {
  label: string;
  name: string;
  placeholder?: string;
  hideArrow?: boolean;
  error?: boolean;
  type: "email" | "password" | "text" | "number" | "tel" | "textarea";
  children?: ReactElement;
}

const FormInput = (props: Props) => {
  const { label, name, placeholder, type, hideArrow, error, children } = props;
  const input = children || (
    <Field
      className={generateClassName("FormInput__Field", [
        hideArrow ? "noArrow" : "",
        error ? "error" : "",
      ])}
      type={type === "textarea" ? "text" : type}
      component={type === "textarea" ? type : null}
      name={name}
      placeholder={placeholder}
    />
  );
  return (
    <div
      className={generateClassName("FormInput", [
        children ? "children" : "",
        error ? "error" : "",
      ])}
    >
      <label className="FormInput__Label" htmlFor={name}>
        {label}
      </label>
      {input}
      <ErrorMessage className="FormInput__Error" name={name} component="div" />
    </div>
  );
};
FormInput.displayName = "FormInput";
export { FormInput };
