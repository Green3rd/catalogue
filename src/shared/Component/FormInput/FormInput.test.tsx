import { shallow } from 'enzyme';
import { FormInput } from './FormInput';
import enzymeToJson from 'enzyme-to-json';

const setup = () => {
  const ownProps = {
    label: 'text',
    name: 'errorText',
    placeholder: 'placeholder',
    type: 'email',
  } as any;

  const props = {
    ...ownProps,
  };

  const Component = shallow(<FormInput {...props} />);
  return { Component, props };
};

describe('<FormInput />', () => {
  it('should render and function correctly', () => {
    const { Component } = setup();
    expect(enzymeToJson(Component)).toMatchSnapshot();
  });
});
