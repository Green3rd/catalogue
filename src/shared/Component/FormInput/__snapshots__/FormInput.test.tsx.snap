// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput /> should render and function correctly 1`] = `
<div
  className="FormInput"
>
  <label
    className="FormInput__Label"
    htmlFor="errorText"
  >
    text
  </label>
  <Field
    className="FormInput__Field"
    name="errorText"
    placeholder="placeholder"
    type="email"
  />
  <FormikConnect(ErrorMessageImpl)
    className="FormInput__Error"
    component="div"
    name="errorText"
  />
</div>
`;
